src/main/java/com/epam/Main.java

```diff
@@ -1,24 +1,178 @@
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
 
+import java.io.File;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
 
 /**
  * Created by Tkachi on 2016/4/3.
  */
 public class Main {
+    private static Properties props = new Properties();
+    private static final String ACCIDENTS_CSV_DIR = "src/main/resources/input";
+
+    private static AtomicInteger filesRead = new AtomicInteger(0);
+    private static AtomicInteger batchesRead = new AtomicInteger(0);
+    private static AtomicInteger batchesProcessed = new AtomicInteger(0);
+    private static AtomicInteger batchesWritten = new AtomicInteger(0);
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        // Load properties
+        props.load(Main.class.getClassLoader().getResourceAsStream("config.properties"));
+        int readingQueueDepth = parseIntProp("reading-queue-depth");
+        int readingThreadsNum = parseIntProp("reading-thread-num");
+        int batchSize = parseIntProp("batch-size");
+
+        int processingQueueDepth = parseIntProp("processing-queue-depth");
+        int processingThreadsNum = parseIntProp("processing-thread-num");
+
+        int writingQueueDepth = parseIntProp("writing-queue-depth");
+        int writingThreadsNum = parseIntProp("writing-thread-num");
+
+        DataProcessor processor = new DataProcessor();
+        List<File> files = Arrays.asList(new File(ACCIDENTS_CSV_DIR).listFiles());
+
+        AccidentsDataWriter dayWriter = new AccidentsDataWriter("day.csv");
+        AccidentsDataWriter nightWriter = new AccidentsDataWriter("night.csv");
 
-    private static final String ACCIDENTS_CSV = "src/main/resources/DfTRoadSafety_Accidents_2009.csv";
+        // Init thread pools
+        ExecutorService readingExecutor = createExecutor("Reading", readingThreadsNum, readingQueueDepth);
+        ExecutorService processingExecutor = createExecutor("Processing", processingThreadsNum, processingQueueDepth);
+        ExecutorService outputExecutor = createExecutor("Writing", writingThreadsNum, writingQueueDepth);
 
+        // Start reading thread(s)
+        files.stream()
+                .forEach(file -> readingExecutor.execute(() -> {
+                    AccidentsDataLoader accidentsDataLoader = null;
+                    try {
+                        accidentsDataLoader = new AccidentsDataLoader(file.getAbsolutePath());
 
-    public static void main(String[] args) throws IOException {
-        AccidentsDataLoader accidentsDataLoader = new AccidentsDataLoader();
-        List<RoadAccident> accidents = accidentsDataLoader.loadRoadAccidents(ACCIDENTS_CSV);
+                        while(accidentsDataLoader.hasMore()) { // Not sure if hasMore() is thread-safe
```

Comment:
this method hasMore() is not thread-safe as it use iterator, however we don't modify roadAccident collection so it is fine.